### ISSUE TITLE
[Vanilla Bug] Fix crash when there are more than 75 cameos in a sidebar strip ("Whiteboy bug")

### ISSUE THREAD

**Issue by Rampastring** [Vinifera-Developers/Vinifera#166]
### Description:
The game only allocates memory for 75 cameos in a sidebar strip. When the number of cameos on the sidebar exceeds 75, memory gets corrupted and the game crashes.

### Additional Files:
CnCNet ts-patches works around this by simply not adding any cameos past 75 to the sidebar: https://github.com/CnCNet/ts-patches/blob/master/src/whiteboy_cameo_bugfix.asm

Naturally this fix is unoptimal and it'd be better if the sidebar just supported more cameos (preferably without any hardcoded limit, storing the cameos in a dynamic vector instead of a static array).

**Comment by CCHyper** [comment]
Agreed, switching the sidebar to a DynamicVectorClass instance would be the ideal solution. It looks like we would need to extend SidebarClass::StripClass and replace all references to the `Buildables` array.
